js/packages/mobile-wallet-adapter-protocol/src/transact.ts

```diff
@@ -17,11 +17,14 @@ const WEBSOCKET_CONNECTION_CONFIG = {
     /**
      * 300 milliseconds is a generally accepted threshold for what someone
      * would consider an acceptable response time for a user interface
-     * after having performed a low-attention tapping task. We set the
+     * after having performed a low-attention tapping task. We set the initial
      * interval at which we wait for the wallet to set up the websocket at
-     * half this, as per the Nyquist frequency.
+     * half this, as per the Nyquist frequency, with a progressive backoff
+     * sequence from there. The total wait time is 30s, which allows for the
+     * user to be presented with a disambiguation dialog, select a wallet, and
+     * for the wallet app to subsequently start.
      */
-    retryDelayMs: 150,
+    retryDelayMs: [150, 150, 200, 500, 500, 750, 750, 1000],
```

Comment:
@steveluscher I observed some interesting behavior here. With this backoff schedule, I observed the following connection attempts (I made a mistake in my log statement format, but the timestamps are still accurate):
<img width="996" alt="Screen Shot 2022-07-23 at 2 08 58 PM" src="https://user-images.githubusercontent.com/103003665/180623120-f31f6be2-c3c8-40f2-9d84-9d0849b07eee.png">

It looks like maybe the web browser is limiting the rate at which the web dapp is allowed to wake up after a timeout occurs. The first few wakeups are roughly on schedule, but after that, I see a sequence of a pretty consistent ~2s wakeup period (twice what was requested), and towards the end 3-6s.

The total wait time for this particular session was 1m14s (vs an expected ~30s).

I wouldn't want to encode this behavior into the web dapps, since it may not be consistent with browser versions, operating modes, etc. However, I'm also not sure what the UX of this extended timeout would be on web dapps. Would love your thoughts here.

Reply:
Totally understand. I'll change this to target a deadline rather than a number of iterations.